std-bits/table/src/main/java/org/enso/table/write/DelimitedWriter.java

```diff
@@ -185,12 +196,10 @@ private String quote(String value, boolean wantsQuoting) {
 
     boolean containsQuote = value.indexOf(quoteChar) >= 0;
     boolean containsQuoteEscape = quoteEscape != null && value.indexOf(quoteEscapeChar) >= 0;
-    boolean shouldQuote =
-        wantsQuoting
-            || containsQuote
-            || containsQuoteEscape
-            || value.indexOf(delimiter) >= 0
-            || value.indexOf(commentChar) >= 0;
+    boolean shouldQuote = wantsQuoting
+        || containsQuote
+        || containsQuoteEscape
+        || containsOtherCharactersThatNeedQuoting(value);
```

Comment:
we could merge this into a single pass check?

Reply:
I wanted to, but I need the `containsQuote` information later on in the script. But I guess I can move the loop into this function and do sth a bit more clever and it will work. Ok will do.